subsys/dfu/img_util/flash_img.c

```diff
@@ -23,8 +23,15 @@ LOG_MODULE_REGISTER(flash_img, CONFIG_IMG_MANAGER_LOG_LEVEL);
 #include <bootutil/bootutil_public.h>
 #endif
 
-#define FIXED_PARTITION_IS_RUNNING_APP_PARTITION(label) \
-	(FIXED_PARTITION_OFFSET(label) == CONFIG_FLASH_LOAD_OFFSET)
+#ifdef CONFIG_USE_DT_CODE_PARTITION
+#define FLASH_LOAD_OFFSET DT_REG_ADDR(DT_CHOSEN(zephyr_code_partition))
+#elif defined(CONFIG_FLASH_LOAD_OFFSET)
+#define FLASH_LOAD_OFFSET CONFIG_FLASH_LOAD_OFFSET
+#endif
```

Comment:
why this, the default of CONFIG_FLASH_LOAD_OFFSET is already the address of the zephyr code partition, when CONFIG_USE_DT_CODE_PARTITION is enabled, see
https://github.com/zephyrproject-rtos/zephyr/blob/8c13c0740c4b67253f8ec812b8d3a5e1d5a7e2e4/Kconfig.zephyr#L102-L113

Reply:
To my understanding, in the ideal case, those are the same, but as tested by this commit: https://github.com/zephyrproject-rtos/zephyr/pull/92038/commits/a84511dd02aab02f9761353ff2eca38889f5721b (stale PR, not merged) it is not always the case.

The logic behind it was that if we enable `CONFIG_USE_DT_CODE_PARTITION`, we should avoid using/relying on the value of `CONFIG_FLASH_LOAD_OFFSET` even if it is a nice shortcut.

The main point of this PR is not about those Kconfigs though, so if you find this change blocking, I will simply remove it. Just let me know what you'd prefer.

Reply:
then cherry pick that commit https://github.com/zephyrproject-rtos/zephyr/commit/a84511dd02aab02f9761353ff2eca38889f5721b Then we don't have to make sure later because there a mismatch will cause a fatal error.

yes blocking